### PR TITLE
tests: fix flaky local deployment test

### PR DESCRIPTION
**Pull request by gssbzn**
Reduce flakyness of the local deployments tests

### CODE REVIEW

File: build/ci/evergreen.yml [hunk 1]
```diff
@@ -201,12 +201,10 @@ functions:
     - command: shell.exec
       type: setup
       params:
-        <<: *go_options
         working_dir: src/github.com/mongodb/mongodb-atlas-cli/bin
         shell: bash
         script: |
-          set -Eeou pipefail
-          export PATH="$ADD_PATH:$PATH"
```
Comment (gssbzn) [inline build/ci/evergreen.yml:209]: unused

File: test/e2e/atlas/deployments_local_test.go [hunk 2]
```diff
@@ -49,9 +49,6 @@ func splitOutput(cmd *exec.Cmd) (string, string, error) {
 func TestDeploymentsLocal(t *testing.T) {
 	const deploymentName = "test"
 
-	g := newAtlasE2ETestGenerator(t)
```
Comment (gssbzn) [inline test/e2e/atlas/deployments_local_test.go:52]: seems we are creating a project just to make sure to list only local deployments, I added an option to filter local only

File: internal/cli/atlas/deployments/list.go [hunk 3]
```diff
@@ -93,11 +95,17 @@ func ListBuilder() *cobra.Command {
...
 		},
 	}
 
 	cmd.Flags().StringVar(&opts.ProjectID, flag.ProjectID, "", usage.ProjectID)
+	cmd.Flags().StringVar(&opts.DeploymentType, flag.TypeFlag, "", usage.DeploymentType)
+
+	_ = cmd.RegisterFlagCompletionFunc(flag.TypeFlag, func(cmd *cobra.Command, args []string, toComplete string) ([]string, cobra.ShellCompDirective) {
+		return options.DeploymentTypeOptions, cobra.ShellCompDirectiveDefault
+	})
+	_ = cmd.Flags().MarkHidden(flag.TypeFlag)
```
Comment (gssbzn) [inline internal/cli/atlas/deployments/list.go:108]: hidden for now since only used for testing

File: build/ci/evergreen.yml [hunk 4]
```diff
@@ -1366,6 +1364,16 @@ tasks:
...
       - func: "install podman"
+      - command: shell.exec
+        type: setup
+        params:
+          shell: bash
+          script: |
+            set -e
+            cat /proc/sys/fs/inotify/max_user_watches
+            cat /proc/sys/fs/inotify/max_user_instances
+            echo 256 | sudo tee /proc/sys/fs/inotify/max_user_instances
```
Comment (gssbzn) [inline build/ci/evergreen.yml:1375]: it seems most of the flakiness was due to too many open files, bumping this for the test

**Review by blva — APPROVED** [review]

**Pull request merged**